### PR TITLE
Fix app iframe resolution

### PR DESCRIPTION
**Pull request by orzechdev**
It fixes the app container to place the app iframe in itself properly.

**PR intended to be tested with API branch:** 

### Pull Request Checklist

1. [ ] This code contains UI changes
2. [ ] All visible strings are translated with proper context including data-formatting
3. [ ] Attributes `[data-test-id]` are added for new elements
4. [ ] Changes are mentioned in the changelog
5. [ ] The changes are tested in different browsers and in light/dark mode

### Test environment config

API_URI=https://automation-dashboard.staging.saleor.cloud/graphql/
MARKETPLACE_URL=https://marketplace-gray.vercel.app/

### CODE REVIEW

File: src/apps/components/AppPage/styles.ts [hunk 1]
```diff
@@ -51,10 +51,15 @@ export const useStyles = makeStyles(
...
     iframeContainer: {
+      lineHeight: 0, // It removes extra space between iframe and container
       "& > iframe": {
         border: "none",
-        minHeight: "75vh",
+        minHeight: "60vh",
+        height: `calc(100vh - ${theme.spacing(34.75)})`,
         width: "100%",
+        [theme.breakpoints.up("md")]: {
+          height: `calc(100vh - ${theme.spacing(29.25)})`,
```
Comment (dominik-zeglen) [inline src/apps/components/AppPage/styles.ts:61]: Just use px/rem value

File: src/hooks/useLocationState.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import { appsSection } from "@saleor/apps/urls";
+import { matchPath, useLocation } from "react-router";
+
+const isAppPath = (pathname: string) =>
+  !!matchPath(pathname, {
+    path: `${appsSection}:id`,
+  });
+
+export const useLocationState = () => {
```
Comment (dominik-zeglen) [inline src/hooks/useLocationState.ts:9]: Can we add jsdoc style comment what is the purpose of this hook?

File: src/components/AppLayout/AppLayout.tsx [hunk 3]
```diff
@@ -117,10 +120,14 @@ const useStyles = makeStyles(
...
+  fullSize?: boolean;
 }
 
-const AppLayout: React.FC<AppLayoutProps> = ({ children }) => {
-  const classes = useStyles({});
+const AppLayout: React.FC<AppLayoutProps> = ({
+  children,
+  fullSize = false,
+}) => {
+  const classes = useStyles({ fullSize });
```
Comment (dominik-zeglen) [inline src/components/AppLayout/AppLayout.tsx:130]: I dont think its used

**Review by dominik-zeglen — CHANGES_REQUESTED** [review]

File: src/hooks/useLocationState.ts [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+
+/*
+ * Use detailed information about the current location.
+ */
+export const useLocationState = () => {
+  const location = useLocation();
+
+  return {
+    isAppPath: isAppPath(location.pathname),
+  };
```
Comment (Droniu) [inline src/hooks/useLocationState.ts:17]: I don't see the generic uses of this hook outside of apps, I'd put it in `src/apps` or `src/apps/utils`

File: src/apps/components/AppPage/styles.ts [hunk 5]
```diff
@@ -51,10 +51,15 @@ export const useStyles = makeStyles(
...
+      lineHeight: 0, // It removes extra space between iframe and container
       "& > iframe": {
         border: "none",
-        minHeight: "75vh",
+        minHeight: "60vh",
+        height: `calc(100vh - 27.8rem)`,
         width: "100%",
+        [theme.breakpoints.up("md")]: {
+          height: `calc(100vh - 23.4rem)`,
+        },
```
Comment (Droniu) [inline src/apps/components/AppPage/styles.ts:62]: This still looks a little weird to me, but if it's usable at 200% zoom then it's fine

**Review by Droniu — APPROVED** [review]

File: src/hooks/useLocationState.ts (hunk 4 above)
Comment (dominik-zeglen) [reply src/hooks/useLocationState.ts:17]: Agreed

File: src/apps/components/AppPage/styles.ts (hunk 5 above)
Comment (orzechdev) [reply src/apps/components/AppPage/styles.ts:62]: It might be usable then

**Review by dominik-zeglen — APPROVED** [review]

**Pull request merged**